src/main/java/com/google/sps/servlets/UserLoginServlet.java

```diff
@@ -34,11 +37,25 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
       String loggedOutUrl = userService.createLogoutURL("/index.html");
       loginInfo.addProperty("LogOutUrl", loggedOutUrl);
       loginInfo.addProperty("LogInUrl", "");
-
+      try {
+        if(SessionContext.getInstance() != null){
+            boolean value = SessionContext.getInstance().isUserExistingInDatastore();
+            loginInfo.addProperty("HasProfile", value);
+        }
+        else {
+            loginInfo.addProperty("HasProfile", false);
+        }
+      
+      } catch(Exception e) {
+        System.err.println("Exception has been caught " + e);
+      }
     } else {
-      String loggedInUrl = userService.createLoginURL("/index.html");
+      String loggedInUrl = userService.createLoginURL("/logged_in_homepage.html");
+   
+    
       loginInfo.addProperty("LogInUrl", loggedInUrl);
       loginInfo.addProperty("LogOutUrl", "");
+      loginInfo.addProperty("HasProfile", false);
```

Comment:
Will L58 overwrite the HasProfile value set in L43?

Reply:
Done